File: 240125/최고의 33위치/best-place-of-33.py
n = int(input())

grid = [
    list(map(int, input().split()))
    for _ in range(n)
]

answer = 0


def search(r, c):
    global answer

    count = 0
    
    for i in range(3):
        for j in range(3):
            count += grid[r + i][c + j]
    
    answer = max(answer, count)


for i in range(n - 2):
    for j in range(n - 2):
        search(i, j)

print(answer)

File: 250716/최고의 33위치/best-place-of-33.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const grid = input.slice(1).map(line => line.split(' ').map(Number));

// Please Write your code here.
function solution() {
    let maxCount = 0;

    function chooseGrid(n) {
        for (let i = 0; i < n - 2; i++) {
            for (let j = 0; j < n - 2; j++) {
                const count = gridSearch(i, j);
                maxCount = Math.max(count, maxCount);
            }
        }
    }

    function gridSearch(r, c) {
        let count = 0;

        for (let i = 0; i < 3; i++) {
            for(let j = 0; j < 3; j++) {
                count += grid[r + i][c + j];
            }
        }

        return count;
    }

    chooseGrid(n);
    return maxCount;
}

console.log(solution());